crates/cow-amm/src/factories.rs

```diff
@@ -0,0 +1,46 @@
+use {
+    crate::Amm,
+    anyhow::Result,
+    contracts::{cow_amm_legacy_helper::Event, CowAmmLegacyHelper},
+    ethcontract::common::DeploymentInformation,
+    shared::impl_event_retrieving,
+};
+
+#[async_trait::async_trait]
+pub trait Deployment: Sync + Send {
+    /// Returns the AMM deployed in the given Event.
+    async fn deployed_amm(&self, helper: &CowAmmLegacyHelper) -> Result<Option<Amm>>;
+}
+
+pub mod legacy {
```

Comment:
Do we plan to add more besides `legacy`?

Reply:
Yes, we'll have to have a separate factory logic for the balancer implementation.

Reply:
I see, I was mostly asking to see if it makes sense to have an own file for this, maybe not now, but when the next ones come it may be clearer that way 🤔 wdyt?

Reply:
Changed the implementation such that we no longer need to have one implementation per factory contract so this got simplified a bit.